app/models/UserAnswers.scala

```diff
@@ -33,7 +33,7 @@ final case class UserAnswers(
                             ) {
 
   def arePagesDefined(pages: Seq[QuestionPage[_]]): Boolean = {
-    !pages.exists(page => checkDataPresent(page).isEmpty)
+    pages.forall(page => checkDataPresent(page).nonEmpty)
```

Comment:
you seem to have accidentally modified arePagesDefined? 

forall returns true if called on an empty sequence - so, `Seq().forall(entry => entry.nonEmpty)` would return true. 
`forall(entry => bool)` is essentially equivalent to `!exists(entry => !bool)`.